### PR TITLE
adding `virsh migrate-setspeed`

### PR DESCRIPTION
**Pull request by r-medina**

### CODE REVIEW

File: libvirt.go [hunk 1]
```diff
@@ -204,6 +204,47 @@ func (l *Libvirt) Events(dom string) (<-chan DomainEvent, error) {
...
+	payload := struct {
+		Padding   [4]byte
+		Domain    Domain
+		Bandwidth int64
+		Flags     uint32
+	}{
+		Padding:   [4]byte{0x0, 0x0, 0x1, 0x0},
+		Domain:    *d,
+		Bandwidth: speed,
+		Flags:     0,
```
Comment (mdlayher) [inline libvirt.go]: Can remove this since it's the zero value anyway.

File: libvirt.go [hunk 2]
```diff
@@ -204,6 +204,47 @@ func (l *Libvirt) Events(dom string) (<-chan DomainEvent, error) {
...
+	}
+
+	buf, err := encode(&payload)
+
```
Comment (mdlayher) [inline libvirt.go]: Yank this blank line.  Look weird.

File: libvirt.go [hunk 3]
```diff
@@ -204,6 +204,47 @@ func (l *Libvirt) Events(dom string) (<-chan DomainEvent, error) {
...
+		return err
+	}
+
+	r := <-resp
+	if r.Status != StatusOK {
+		return decodeError(r.Payload)
+	}
+
+	return nil
+
```
Comment (mdlayher) [inline libvirt.go]: Yank this blank line.  Look weird.

File: libvirttest/libvirt.go [hunk 4]
```diff
@@ -146,6 +146,16 @@ var testRunReplyFail = []byte{
 	0x6e, 0x64, 0x22, 0x7d, 0x7d, 0x00, 0x00, 0x00,
 }
 
+var testSetSpeedReply = []byte{
+	0x00, 0x00, 0x00, 0x1c, // length
+	0x20, 0x00, 0x80, 0x86, // program
+	0x00, 0x00, 0x00, 0x01, // version
+	0x00, 0x00, 0x00, 0xcf, // procedure
+	0x00, 0x00, 0x00, 0x01, // type
+	0x00, 0x00, 0x00, 0x05, // serial
```
Comment (benlemasurier) [inline libvirttest/libvirt.go]: Serial should be `0x00` (probably why the test is failing)

File: libvirttest/libvirt.go [hunk 5]
```diff
@@ -248,8 +258,8 @@ func (m *MockLibvirt) handleRemote(procedure uint32, conn net.Conn) {
 		conn.Write(m.reply(testVersionReply))
 	case constants.ProcDomainLookupByName:
 		conn.Write(m.reply(testDomainResponse))
-	case constants.ProcConnectListAllDomains:
-		conn.Write(m.reply(testDomainsReply))
```
Comment (benlemasurier) [inline libvirttest/libvirt.go]: Did you mean to pull this?

Comment (mdlayher) [inline libvirttest/libvirt.go]: Aha.  Good call.

**Comment by benlemasurier** [comment]
👍

**Pull request merged**